### DISCUSSION TITLE
Network porosity crosses 1 using SNOW

### DISCUSSION THREAD

**Discussion by kunal0kumar0sinha** [PMEAL/porespy#960] [Q&A]
Hi

I am trying to extract a network from a fibrous mat using snow. When I calculate the porosity of the network just considering throats, it crosses 1, which should not be the case. Please help.

dim=200
im = ps.generators.cylinders(shape=[dim, dim, dim], r=5, ncylinders=200)
snow_output = ps.networks.snow2(im, voxel_size=1)
pn = op.io.network_from_porespy(snow_output.network)
snow_output = ps.networks.snow2(im, voxel_size=1)
pn = op.io.network_from_porespy(snow_output.network)

_Then I calculate porosity with throats only, whenever it is >1 

dt=pn['throat.inscribed_diameter']         #inscribed throat diameter
l=pn['throat.direct_length']               # throat direct length
R=dt/2
por=np.sum(np.pi*R**2*l)/dim**3

Result comes: 1.366_

**Comment by jgostick** [comment]
It is easy to make mistakes when computing porosity in pore networks, especially for high porosity and anisotropic materials (like fibers).  The main problem is that throats can overlap each other, but you must also think about the intersection between pores and throats to make sure they don't get counted twice.
